test/Microsoft.SqlTools.ServiceLayer.TestEnvConfig/Program.cs

```diff
@@ -94,22 +94,24 @@ TestEnvConfig setting_file
 
         private static void SaveSettings(string settingFile)
         {
-            
             var xdoc = XDocument.Load(settingFile);
-            var settings =
-                from setting in xdoc.Descendants("Instance")
-                select new InstanceInfo(setting.Attribute("VersionKey").Value) // VersionKey is required
+            List<InstanceInfo> settings = new List<InstanceInfo>();
+            foreach (var setting in xdoc.Descendants("Instance"))
+            {
+                var passwordEnvVariableValue = Environment.GetEnvironmentVariable((setting.Attribute("VersionKey").Value + "_password"));
+
+                settings.Add(new InstanceInfo(setting.Attribute("VersionKey").Value)
                 {
                     ServerName = setting.Element("DataSource").Value, // DataSource is required
                     ConnectTimeoutAsString = (string)setting.Element("ConnectTimeout"), //ConnectTimeout is optional
-                    User = (string)setting.Element("UserId"), // UserID is optional
-                    Password = (string)setting.Element("Password"),
+                     User = (string)setting.Element("UserId"), // UserID is optional
+                    Password = passwordEnvVariableValue != "" ? passwordEnvVariableValue : (string)setting.Element("Password"), 
```

Comment:
You need to use `string.IsNullOrEmpty` here - `GetEnvironmentVariable` returns null if the env var doesn't exist

https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-6.0